service/create/service.go

```diff
@@ -315,6 +325,21 @@ func allExistingInstancesMatch(instances *ec2.DescribeInstancesOutput, state EC2
 	return true
 }
 
+func (s *Service) uploadCloudconfigToS3(awsSession *awssession.Session, s3Bucket, path, data string) error {
+	svc := s3.New(awsSession)
```

Comment:
This might be me nitpicking. I think all kind of client initialization should be done during program initialization. I feel like this is more about actual runtime.